Mathlib/Data/Int/Card.lean

```diff
@@ -0,0 +1,140 @@
+/-
+Copyright (c) 2024 Jeremy Tan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Tan
+-/
+import Mathlib.Data.Int.Interval
+import Mathlib.Data.Int.ModEq
+import Mathlib.Data.Nat.Count
+import Mathlib.Data.Nat.Interval
+import Mathlib.Data.Rat.Floor
+
+/-!
+# Counting elements in an interval with given residue
+
+The theorems in this file generalise `card_multiples` in `Data.Nat.Factorization.Basic` to
+all integer intervals and any fixed residue (not just zero, which reduces to the multiples).
+Theorems are given for `Ico` and `Ioc` intervals.
+-/
+
+
+open Finset Int
+
+namespace Int
+
+variable (a b : ℤ) {r : ℤ} (hr : 0 < r)
+
+lemma Ico_filter_dvd_eq : (Ico a b).filter (r ∣ ·) =
+    (Ico ⌈a / (r : ℚ)⌉ ⌈b / (r : ℚ)⌉).map ⟨_, mul_left_injective₀ hr.ne'⟩ := by
```

Comment:
```suggestion
    (Ico ⌈a / (r : ℚ)⌉ ⌈b / (r : ℚ)⌉).map ⟨(· * r), mul_left_injective₀ hr.ne'⟩ := by
```
It doesn't help Lean, but it definitely helps the reader understand what the theorem is saying.

Same thing below several times.